File: capacitor.java
  

/**---------------------------------------------------
 * Akdeniz University CSE122 Physics II Homework
 * @author: Yahya Efe Kuruçay
 * @since: 26.05.2024
 * Description: Homework
 * Score: ?
 * Website: https://efekurucay.com
*---------------------------------------------------*/


/***
 *    ███████ ███████ ███████   |    ███████ ███████ ███████ 
 *    ██      ██      ██        |    ██      ██      ██      
 *    █████   █████   █████     |    █████   █████   █████   
 *    ██      ██      ██        |    ██      ██      ██      
 *    ███████ ██      ███████   |    ███████ ██      ███████ 
 *                            
 *                            
 */



import java.util.Scanner;

public class capacitor {
    // epsilon (F/m)
    private static final double EPSILON_0 = 8.854e-12;

    // Kapasitans hesaplama
    public static double calculateCapacitance(double area, double separation) {
        double area_m2 = area * 1e-6; // mm^2'yi m^2'ye dönüştür
        double separation_m = separation * 1e-3; // mm'yi m'ye dönüştür
        double capacitance = (EPSILON_0 * area_m2) / separation_m;
        return capacitance * 1e12; // F'yi pF'ye dönüştür
    }

    // Üst plakada biriken yük yükü 
    public static double calculateCharge(double capacitance, double voltage) {
        return capacitance * voltage; // Yük (pC)
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter plate area (mm^2): ");
        double area = scanner.nextDouble();

        System.out.print("Enter separation (mm): ");
        double separation = scanner.nextDouble();

        System.out.print("Enter battery voltage (V): ");
        double voltage = scanner.nextDouble();

        // Kapasitansı hesapla
        double capacitance = calculateCapacitance(area, separation);

        // Üst plaka yükünü hesapla
        double charge = calculateCharge(capacitance, voltage);

        // Sonuçları ekrana yazdır
        System.out.printf("Capacitance: %.2f pF%n", capacitance);
        System.out.printf("Top Plate Charge: %.2f pC%n", charge);

        scanner.close();
    }
}

/***
 *              _____                    _____                    _____          
 *             /\    \                  /\    \                  /\    \         
 *            /::\    \                /::\    \                /::\    \        
 *           /::::\    \              /::::\    \              /::::\    \       
 *          /::::::\    \            /::::::\    \            /::::::\    \      
 *         /:::/\:::\    \          /:::/\:::\    \          /:::/\:::\    \     
 *        /:::/__\:::\    \        /:::/__\:::\    \        /:::/__\:::\    \    
 *       /::::\   \:::\    \      /::::\   \:::\    \      /::::\   \:::\    \   
 *      /::::::\   \:::\    \    /::::::\   \:::\    \    /::::::\   \:::\    \  
 *     /:::/\:::\   \:::\    \  /:::/\:::\   \:::\    \  /:::/\:::\   \:::\    \ 
 *    /:::/__\:::\   \:::\____\/:::/  \:::\   \:::\____\/:::/__\:::\   \:::\____\
 *    \:::\   \:::\   \::/    /\::/    \:::\   \::/    /\:::\   \:::\   \::/    /
 *     \:::\   \:::\   \/____/  \/____/ \:::\   \/____/  \:::\   \:::\   \/____/ 
 *      \:::\   \:::\    \               \:::\    \       \:::\   \:::\    \     
 *       \:::\   \:::\____\               \:::\____\       \:::\   \:::\____\    
 *        \:::\   \::/    /                \::/    /        \:::\   \::/    /    
 *         \:::\   \/____/                  \/____/          \:::\   \/____/     
 *          \:::\    \                                        \:::\    \         
 *           \:::\____\                                        \:::\____\        
 *            \::/    /                                         \::/    /        
 *             \/____/                                           \/____/         
 *                                                                               
 */

/*
 *            __     _                                                          
 *           / _|   | |                                                         
 *       ___| |_ ___| | ___   _ _ __ _   _  ___ __ _ _   _   ___ ___  _ __ ___  
 *      / _ \  _/ _ \ |/ / | | | '__| | | |/ __/ _` | | | | / __/ _ \| '_ ` _ \ 
 *     |  __/ ||  __/   <| |_| | |  | |_| | (_| (_| | |_| || (_| (_) | | | | | |
 *      \___|_| \___|_|\_\\__,_|_|   \__,_|\___\__,_|\__, (_)___\___/|_| |_| |_|
 *                                                    __/ |                     
 *                                                   |___/                      
 */

File: capacitor.py
import tkinter as tk
from tkinter import ttk

# Kapasitans hesaplama fonksiyonu
def calculate_capacitance(area, separation, voltage):
    epsilon_0 = 8.854e-12  # Boşluğun elektriksel geçirgenliği (F/m)
    area_m2 = area * 1e-6  # mm^2'yi m^2'ye dönüştürme
    separation_m = separation * 1e-3  # mm'yi m'ye dönüştürme
    capacitance = (epsilon_0 * area_m2) / separation_m
    return capacitance * 1e12  # F'yi pF'ye dönüştürme

# Track bar değerlerinin değişiminde çağrılan fonksiyon
def update_values(event=None):
    area = area_scale.get()
    separation = separation_scale.get()
    voltage = voltage_scale.get()
    
    capacitance = calculate_capacitance(area, separation, voltage)
    charge = capacitance * voltage  # Q = C * V
    
    area_value_label.config(text=f"Plate Area: {area:.1f} mm²")
    separation_value_label.config(text=f"Separation: {separation:.1f} mm")
    voltage_value_label.config(text=f"Battery Voltage: {voltage:.1f} V")
    capacitance_label.config(text=f"Capacitance: {capacitance:.2f} pF")
    charge_label.config(text=f"Top Plate Charge: {charge:.2f} pC")

# Tkinter arayüzü
root = tk.Tk()
root.title("Capacitor Calculation")

# Track barlar
area_scale = tk.Scale(root, from_=100, to_=1000, resolution=0.1, orient="horizontal", command=update_values)
area_scale.set(100)
area_scale.pack()
area_value_label = tk.Label(root, text="Plate Area: 100.0 mm²")
area_value_label.pack()

separation_scale = tk.Scale(root, from_=1, to_=10, resolution=0.1, orient="horizontal", command=update_values)
separation_scale.set(2)
separation_scale.pack()
separation_value_label = tk.Label(root, text="Separation: 2.0 mm")
separation_value_label.pack()

voltage_scale = tk.Scale(root, from_=1, to_=10, resolution=0.1, orient="horizontal", command=update_values)
voltage_scale.set(1.5)
voltage_scale.pack()
voltage_value_label = tk.Label(root, text="Battery Voltage: 1.5 V")
voltage_value_label.pack()

# Kapasitans ve üst plaka yükü etiketleri
capacitance_label = tk.Label(root, text="Capacitance: ")
capacitance_label.pack()

charge_label = tk.Label(root, text="Top Plate Charge: ")
charge_label.pack()

# Başlangıç değerlerini güncelle
update_values()

# Arayüzü başlat
root.mainloop()
